packages/webcomponents/src/components/checkout-wrapper/checkout-wrapper.tsx

```diff
@@ -44,44 +44,52 @@ export class CheckoutWrapper {
     this.queryFormRefs();
   }
 
+  disconnectedCallback() {
+    this.observer?.disconnect();
+  }
+
   private queryFormRefs() {
-    this.cardFormRef = this.hostEl.querySelector('justifi-card-form');
-    this.bankAccountFormRef = this.hostEl.querySelector('justifi-bank-account-form');
+    this.paymentMethodFormRef = this.hostEl.querySelector('justifi-card-form, justifi-bank-account-form');
+    this.billingInformationFormRef = this.hostEl.querySelector('justifi-billing-information-form, justifi-postal-code-form');
   }
 
   @Method()
   async validate(): Promise<boolean> {
-    if (this.cardFormRef || this.bankAccountFormRef) {
-      const formToValidate = this.cardFormRef || this.bankAccountFormRef;
-      return formToValidate.validate();
-    } else {
-      throw new Error('CardForm component not found');
-    }
+    const validationResults = await Promise.all([
+      this.paymentMethodFormRef?.validate(),
+      this.billingInformationFormRef?.validate()
+    ]);
+
+    return validationResults.every(result => result?.isValid !== false);
   }
 
   @Method()
   async tokenizePaymentMethod(billingInfo: BillingInfo): Promise<any> {
-    if (this.cardFormRef || this.bankAccountFormRef) {
-      const formToTokenize = this.cardFormRef || this.bankAccountFormRef;
-      const isValid = await formToTokenize.validate();
-      if (!isValid) {
-        throw new Error('Card form is not valid');
-      }
-      return formToTokenize.tokenize({
-        clientId: this.authToken,
-        paymentMethodMetadata: {
-          accountId: this.accountId,
-          ...billingInfo
-        },
-        account: this.accountId,
-      });
-    } else {
-      this.errorEvent.emit({
-        message: "Component not found: 'justifi-card-form' or 'justifi-bank-account-form' is required for tokenization.",
+    if (!this.paymentMethodFormRef) {
+      return this.errorEvent.emit({
+        message: 'Payment method form not found',
         errorCode: ComponentErrorCodes.TOKENIZE_ERROR,
         severity: ComponentErrorSeverity.ERROR,
       });
     }
+
+    const isValid = await this.validate();
+    if (!isValid) throw new Error('Form is not valid');
```

Comment:
I think as long as client-side errors are shown upon validation, we should just return here and not throw an error. Especially since we have the event emitting already